HW2/P2/parallel_vector.pyx

```diff
@@ -74,21 +74,21 @@ cpdef move_data_fine_grained(np.int32_t[:] counts,
        int idx, r
        omp_lock_t *locks = get_N_locks(counts.shape[0])
 
-   ##########
-   # Your code here
-   # Use parallel.prange() and a lock for each element of counts to parallelize
-   # data movement.  Be sure to avoid deadlock, and double-locking.
-   ##########
    with nogil:
-       for r in range(repeat):
-           for idx in range(src.shape[0]):
+      for r in range(repeat):
+         for idx in prange(src.shape[0], num_threads = 4):
+            if src[idx] != dest[idx]:   
                if counts[src[idx]] > 0:
```

Comment:
Oh, that makes a lot of sense. I was having trouble thinking about it, so this really helps. Thanks.